### PR TITLE
1.21: Update to Berkeley DB 5.3

### PR DESCRIPTION
**Pull request by patricklodder**
Re-integration of #2245.

I took the original commits from @bjacquin and @rnicoll and re-did the CI integration previously done with [`a404897`](https://github.com/dogecoin/dogecoin/pull/2245/commits/a4048978eb79f9c03a289ae1785c0146ff8e9f6d), because the configuration there was incomplete.

Doing the bdb 5.3 switch now will allow us to spot any regressions from a wallet perspective when we fully enable auxpow on 1.21-dev, as 1.14 wallets files are incompatible with the bdb 4.8 currently used.

Notes:

- I kept the `--with-incompatible-bdb` flag in the centos CI for now because I'm not sure whether it's good or bad to test the scenario. Open to discussion.
- ~~It would be good if we can attempt a gitian/guix build & compare with this, but per [Ross' comment back in November](https://github.com/dogecoin/dogecoin/pull/2245#issuecomment-983052757), it's currently broken. If that's still the case, I think we should address that separately.~~

### CODE REVIEW

File: doc/build-netbsd.md [hunk 1]
```diff
@@ -35,16 +35,16 @@ BerkeleyDB is only necessary for the wallet functionality. To skip this, pass
 It is recommended to use Berkeley DB 4.8. You cannot use the BerkeleyDB library
```
Comment (chromatic) [inline doc/build-netbsd.md:35]: Looks like this line needs an update too.

Do we know if the bdb library in ports still has this limitation? (That's question shouldn't block this PR.)

File: doc/build-openbsd.md [hunk 2]
```diff
@@ -35,16 +35,16 @@ BerkeleyDB is only necessary for the wallet functionality. To skip this, pass
 It is recommended to use Berkeley DB 4.8. You cannot use the BerkeleyDB library
```
Comment (chromatic) [inline doc/build-openbsd.md:35]: Similar question here.

**Comment by patricklodder** [comment]
> It would be good if we can attempt a gitian/guix build & compare with this, but per https://github.com/dogecoin/dogecoin/pull/2245#issuecomment-983052757, it's currently broken. If that's still the case, I think we should address that separately.

I'm dropping this for now, we can work on deterministic builds under #2852, since it looks like both gitian and guix need some work.

**Comment by patricklodder** [comment]
@chromatic I went over all the docs that still referenced either the need to install 4.8, or were describing processes that were not valid for the 5.3 installation, and changed these.

Note that some of the installation instructions don't fully work yet, but I would like to keep that out of the scope of this PR - we can work on that later.

**Review by chromatic — APPROVED** [review]
ACK. Tested on 64-bit Linux. The documentation changes look good, and I'm comfortable addressing Gitian/Guix separately.

**Pull request merged**